langchain_postgres/vectorstores.py

```diff
@@ -2157,6 +2282,64 @@ async def amax_marginal_relevance_search_by_vector(
 
         return _results_to_docs(docs_and_scores)
 
+    def _prepare_create_hnsw_index_query(
+        self,
+        dims: int = ADA_TOKEN_COUNT,
+        distance_strategy: HNSWDistanceStrategy = DEFAULT_HNSW_DISTANCE_STRATEGY,
+        m: int = 8,
+        ef_construction: int = 16,
+    ) -> sqlalchemy.TextClause:
+        create_index_query = sqlalchemy.text(
+            "CREATE INDEX IF NOT EXISTS langchain_pg_embedding_idx "
+            "ON langchain_pg_embedding USING hnsw ((embedding::vector({})) {}) "
+            "WITH ("
+            "m = {}, "
+            "ef_construction = {}"
+            ");".format(dims, distance_strategy, m, ef_construction)
+        )
+
+        return create_index_query
+
+    def create_hnsw_index(
```

Comment:
Is there anyway to move this to a classmethod, so we can create the ahead of time.

I want to create a control plane -- that allows  managing the indexes without having to actually instantiate an index.


```python
class IndexManager:
  def list_indexes(**kwargs) -> List[Dict]:
  
def get_index(id) -> VectorStore:
   pass

 def create_index(**kwargs) -> str:
    pass
```

Reply:
Hi @eyurtsev  thanks for the feedback. Honestly I'm not too sure how to do this as this is my first PR to the project. Feel free to commandeer the PR (or merge it and create a new one if it makes more sense)